tests/remote/test_thunderbird.py

```diff
@@ -4,248 +4,124 @@
+    """Test release scraper against the remote scraper."""
+    scraper = mozdownload.ReleaseScraper(destination=tmpdir, **args)
+
+    if url:
+        assert urllib.unquote(scraper.url) == urljoin(BASE_URL, url)
+
+@pytest.mark.parametrize("args,url", [
+    ({'application': 'thunderbird', 'platform': 'linux', 'version': '52.7.0'},
+     'thunderbird/candidates/52.7.0-candidates/build1/linux-i686/en-US/thunderbird-52.7.0.tar.bz2'),
+    ({'application': 'thunderbird', 'platform': 'linux64', 'version': '52.7.0'},
+     'thunderbird/candidates/52.7.0-candidates/build1/linux-x86_64/en-US/thunderbird-52.7.0.tar.bz2'),
+    ({'application': 'thunderbird', 'platform': 'mac', 'version': '52.7.0'},
+     'thunderbird/candidates/52.7.0-candidates/build1/mac/en-US/Thunderbird 52.7.0.dmg'),
+    ({'application': 'thunderbird', 'platform': 'win32', 'version': '52.7.0', 'locale': 'en-GB'},
+     'thunderbird/candidates/52.7.0-candidates/build1/win32/en-GB/Thunderbird Setup 52.7.0.exe'),
+])
+def test_candidate_scraper(tmpdir, args, url):
+    """Test release candidate scraper against the remote server."""
+    scraper = mozdownload.ReleaseCandidateScraper(destination=tmpdir, **args)
+
+    assert urllib.unquote(scraper.url) == urljoin(BASE_URL, url)
+
+@pytest.mark.parametrize("args", [
+    ({'application': 'thunderbird',
+      'platform': 'linux',
+      'branch': 'comm-central'}),
+    ({'application': 'thunderbird',
+      'platform': 'linux',
+      'branch': 'comm-central',
+      'build_id': '20180301030201',
+      'extension': 'txt'}),
+    ({'application': 'thunderbird',
+      'platform': 'linux',
+      'branch': 'comm-central',
+      'build_id': '20180301030201',
+      'locale': 'de'}),
+    ({'application': 'thunderbird',
+      'platform': 'linux64',
+      'branch': 'comm-central'}),
+    ({'application': 'thunderbird',
+      'platform': 'mac',
+      'branch': 'comm-central'}),
+    ({'application': 'thunderbird',
+      'platform': 'win32',
+      'branch': 'comm-central'}),
+    ({'application': 'thunderbird',
+      'platform': 'win64',
+      'branch': 'comm-central'}),
+    ({'application': 'thunderbird',
+      'platform': 'win64',
+      'branch': 'comm-central',
+      'date': '2018-03-01'}),
+    ({'application': 'thunderbird',
+      'platform': 'win64',
+      'branch': 'comm-central',
+      'date': '2018-03-01',
+      'build_number': 1}),
+    ({'application': 'thunderbird',
+      'platform': 'win64',
+      'branch': 'comm-central',
+      'build_id': '20180301030201'}),
+])
+def test_daily_scraper(tmpdir, args):
+    """Test daily scraper against the remote server."""
+    mozdownload.DailyScraper(destination=tmpdir, **args)
+
+@pytest.mark.parametrize("args", [
+    ({'application': 'thunderbird',
+      'platform': 'linux',
+      'branch': 'comm-central'}),
+    ({'application': 'thunderbird',
+      'platform': 'linux',
+      'branch': 'comm-central',
+      'extension': 'txt'}),
```

Comment:
That's a special test for the `.txt` extension. Please move this to the very end.